From f79731ff2ae564c960f9bd3bb32dfdfc9627b99b Mon Sep 17 00:00:00 2001
From: Weian Chen <weian.chen@intel.com>
Date: Tue, 30 Apr 2013 13:21:29 +0800
Subject: Add 100us scheduling interval between consecutive video events

BZ: 48604

original patch:
R2: 17284 (BZ 7871 - Media player won't be suspended after press the power button sometimes)
R3: 30123 (BZ 18523)

Add 100us scheduling interval between consecutive video events so that
any attempts to cancel future video events has opportunity to be invoked
during the 100us time slot. For BZ7871, after adding this patch, the
suspend() methods could be invoked successfully after power button
is pressed.

Category: aosp improvement
Domain: Video.Media-local playback
Origin: internal
Upstream-Candidate: yes

Change-Id: Iddee992de718d39966ddf646f1805c49266620ca
Orig-Change-Id: Ida2469d198684409364f72de163d5de022ed5185
Signed-off-by: Weian Chen <weian.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index 9fc9717..130fe94 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -1933,7 +1933,10 @@ void AwesomePlayer::onVideoEvent() {
                 mSeeking = SEEK_VIDEO_ONLY;
                 mSeekTimeUs = mediaTimeUs;
 
-                postVideoEvent_l();
+                // the next video event scheduling will occur after 100us so that
+                // any attempts to cancel future video events could take effect within
+                // this 100us interval
+                postVideoEvent_l(100);
                 return;
             } else {
                 // The widevine extractor doesn't deal well with seeking
@@ -1964,7 +1967,7 @@ void AwesomePlayer::onVideoEvent() {
                     ++mStats.mNumVideoFramesDropped;
                 }
 
-                postVideoEvent_l(0);
+                postVideoEvent_l(100);
                 return;
             }
         }
-- 
1.7.4.1

